drivers/iio/dac/ti-dac5571.c

```diff
@@ -402,6 +402,7 @@ MODULE_DEVICE_TABLE(of, dac5571_of_id);
 #endif
 
 static const struct i2c_device_id dac5571_id[] = {
+	{"dac081c081", single_8bit},
```

Comment:
I don't think this is the right fix. The root cause of the issue is that matching a device to a driver using DT supports the compatible string fallback mechanism, while the i2c_device_id requires an exact match of the first compatible string.

Ideally this should be handled in the I2C core, but that will likely require long discussions. As a short term fix in this driver, I believe it would be better to add the chip_id in the .data field of the of_device_id entries in the dac5571_of_id array, and use that (with a call to of_device_get_match_data()) in the OF case.

Would you like to give it a try, or should I do it ?